### DISCUSSION TITLE
Disabling automatic addition of "use" at the top of the file

### DISCUSSION THREAD

**Discussion by Rafagd** [phpactor/phpactor#2349] [Q&A]
I am working in an old codebase and we don't use "use" here unless it's really needed.
Therefore, I don't want the LSP to add use for the files basically never.

I tried setting completion_worse.completor.use.enabled=false, but it seems like this is not the option, as the LSP still does it.
Does phpactor started via mason even care about the ~/.config/phpactor/phpactor.yml file?

Is there a way to do it?

**Comment by dantleech** [comment]
`completion_worse.completor.use.enabled` is to enable completion of `use ` statements.

there is currently no way to disable auto-import, I would guess that instead you'd want `To\Expand\The\ClassName`?

i'd be open to adding an option to disable auto-import and insert the fully expanded class name instead, although it's quite a niche requirement.

**Comment by Rafagd** [comment]
Yeah, auto-completing the fully-qualified name would be nice.
 I could roll my own personal implementation as well, if you could point me the places where the magic is happening.

**Reply by dantleech** [reply]
If you're brave: https://github.com/phpactor/phpactor/blob/f10652ef66ca06aab0bf46744ed9fe39c2af609a/lib/Extension/LanguageServerCompletion/Handler/CompletionHandler.php#L85

**Comment by Rafagd** [comment]
Thanks, I'll see if I can cook something and I'll pull request if it's not too McGuyvery
